maro/simulator/core.py

```diff
@@ -286,31 +278,30 @@ def _simulate(self):
                 if actions is not None and not isinstance(actions, Iterable):
                     actions = [actions]
 
-                # Generate a new atom event first.
-                action_event = self._event_buffer.gen_action_event(self._tick, actions)
-
-                # NOTE: decision event always be a CascadeEvent
-                # We just append the action into sub event of first pending cascade event.
-                pending_events[0].state = EventState.EXECUTING
-                pending_events[0].add_immediate_event(action_event, is_head=True)
-
-                if self._decision_mode == DecisionMode.Joint:
-                    # For joint event, we will disable following cascade event.
-
-                    # We expect that first action contains a src_event_id to support joint event with sequential action.
-                    action_related_event_id = None if len(
-                        actions) == 1 else getattr(actions[0], "src_event_id", None)
-
-                    # If the first action has a decision event attached, it means sequential action is supported.
-                    is_support_seq_action = action_related_event_id is not None
-
-                    if is_support_seq_action:
-                        for i in range(1, pending_event_length):
-                            if pending_events[i].id == actions[0].src_event_id:
-                                pending_events[i].state = EventState.FINISHED
-                    else:
-                        for i in range(1, pending_event_length):
-                            pending_events[i].state = EventState.FINISHED
+                if self._decision_mode == DecisionMode.Sequential:
+                    # Generate a new atom event first.
+                    action_event = self._event_buffer.gen_action_event(self._tick, actions)
```

Comment:
for Sequential mode, the **actions** is a List?

Reply:
just wrap it as a list, make it same with joint mode